Downloading images from S3 bucket and resizing them

kriti0111 opened the pull request:
Downloading images from S3 bucket and resizing them using Image crate

kriti0111 committed:
Downloading images from S3 bucket and resizing them

Downloading images from S3 bucket and resizing them using Image crate

kahlil29 commented on src/main.rs:
```diff
@@ -1,4 +1,51 @@
 fn main(){
-        println!("hello world");
+        println!("Downloading file from S3 bucket...");
+        let s3 = S3Client::new(Region::ApSoutheast1);
```
Add the region name as a `const` along with the bucket name and image name. 
See [here](https://github.com/Holmusk/rust-foodScoring-service/blob/5f6f0264936e67f9abec1505e956c552b5ba578c/src/rating_utils.rs#L88) for reference on how to parse the region name.

kahlil29 commented on src/main.rs:
```diff
@@ -1,4 +1,51 @@
+           
+    }
+
+    fn resize_image(bytes_img : &[u8]){
```
I think this function should return the same type that it takes as input

kahlil29 reviewed (changes requested):
Let's move the code into a file named `img_resizing_lambda.rs`. 
Also see if you can get this formatter running for the project: https://github.com/rust-lang/rustfmt

kriti0111 committed:
Updated with changes requested

Updated with review comments.
1. Moved the code to a new file
2. Converted resized/scaled image to bytes
3. Parsing fn for AWS Region

kahlil29 commented on src/img_resizing_lambda.rs:
```diff
@@ -0,0 +1,71 @@
+    download_img_from_s3(s3, BUCKET_NAME.to_string(), IMAGE_NAME.to_string());
+}
+
+fn get_region(aws_region_name : String) -> Region{
```
Can you test this function locally please? 
I think the region string input should be `ap-southeast-1`

kriti0111 replied:
Tested this function by printing the region value in Ok as below - 
```
        match aws_region_name.parse::<Region>() {
            Ok(valid_region) => {println!("valid region {:?}",valid_region);
            valid_region
            } ,
            // Default fallback Region (Singapore)
            Err(_) =>  Region::ApSoutheast1,
        }
```
Giving string input(aws_region_name) as `ap-southeast-1' and 'ApSoutheast1' works.
